src/architecture/serialization.md

```diff
@@ -1,533 +1,193 @@
-
-### Protocol Version
-
-The major version of the Exonum serialization protocol. Currently, `0`.
-
-**Binary presentation:** `u8` (unsigned 1-byte integer).  
-**JSON presentation:** number.
+Besides the description of structures, the `.rs` files also contain additional
+code and functions required for protobuf serialization and deserialization.
 
-### Service ID
+For convenience, the [protobuf descriptions][proto-files] of the typical
+structures used in Exonum are already included in the framework.
 
-Used when the message is a transaction.
-Sets the [service](services.md) that a transaction belongs to.
-The pair `(service_id, message_id)` is
-used to look up the implementation of [the transaction interface](transactions.md#interface)
-(e.g., `verify` and `execute` methods).
+## Building Exonum with Protobuf Serialization
 
-**Binary presentation:** `u16` (unsigned 2-byte integer).  
-**JSON presentation:** number.
+Exonum includes the
+[`exonum_build`](https://github.com/exonum/exonum/tree/master/exonum_build)
+crate which lets users add the `protobuf_generate` function to their
+`build.rs`. This function automatically generates the `.rs` files for all the
+`.proto` files during the build process. To use `protobuf_generate`, add the
+following code to `build.rs` indicating the folder which contains the `.proto`
+files:
 
-### Message ID
-
-`message_id` defines the type of message within the service.
-
-!!! note "Example"
-    [The sample cryptocurrency service][cryptocurrency] includes 2 main
-    types of transactions: `TxCreateWallet` for creating a wallet with an initial
-    coins balance,
-    and `TxTransfer` for coin transfer.
-
-**Binary presentation:** `u16` (unsigned 2-byte integer).  
-**JSON presentation:** number.
+```rust
+use exonum_build::protobuf_generate;
 
-### Payload length
+protobuf_generate("src/proto", &["src/proto"], "example_mod.rs")
+```
 
-The length of the entire message serialization (including its header and signature).
+To use protobuf-generated Rust structures, users first need to create a module
+which will include the protobuf-generated files:
 
-**Binary presentation:** `u32` (unsigned 4-byte integer).  
-**JSON presentation:** (not serialized).
+```rust
+extern crate exonum;
 
-### Body
+include!(concat!(env!("OUT_DIR"), "/example_mod.rs"));
 
-Serialized [structure](#structures) (including its header and body) described on
-`transactions!` macro call.
+// If you use types from `exonum` .proto files.
+use exonum::proto::schema::*;
+```
 
-### Signature
+Then to access a structure from a protobuf-generated file, you need to indicate
+the name of the module with the generated files and the name of the required
+file and structure in it:
 
-[Ed25519 digital signature](https://ed25519.cr.yp.to/) over the binary
-serialization of the message (excluding the signature bytes,
-i.e., the last 64 bytes of the serialization).
+```rust
+module_name::proto_file_name::StructNameInProtoFile
```

Comment:
Nit: This isn't valid Rust on its own; I'd prefer a more comprehensive example.

Reply:
Maybe something like this?
```rust
use module_name::proto_file_name::StructName;

fn deserialize(bytes: &[u8]) -> StructName {
    let mut pb = StructName::new();
    pb.merge_from_bytes(slice).unwrap();
    pb
}

fn serialize(pb: &StructName) -> Vec<u8> {
    pb.write_to_bytes().unwrap()
}
```